packages/protocol/contracts/lib/Checkpointing.sol

```diff
@@ -121,18 +127,19 @@ library Checkpointing {
 
         // If the requested time is equal to or after the time of the latest registered value, return latest value
         uint256 lastIndex = length - 1;
-        if (_time >= self.history[lastIndex].time) {
-            return uint256(self.history[lastIndex].value);
+        Checkpoint storage lastCheckpoint = _self.history[lastIndex];
+        if (_time >= lastCheckpoint.time) {
+            return uint256(lastCheckpoint.value);
         }
 
         // If the requested time is previous to the first registered value, return zero to denote missing checkpoint
-        if (_time < self.history[0].time) {
+        if (length == 1 || _time < self.history[0].time) {
             return 0;
         }
 
         // Execute a binary search between the checkpointed times of the history
         uint256 low = 0;
-        uint256 high = lastIndex;
+        uint256 high = lastIndex - 1;
```

Comment:
What’s the rationale for this change?

Reply:
Not a big optimization, but we already checked for the last value above, so there was no point in including it in the binary search. Worst case it's the same number of traversals and storage loads, best case it can reduce the search depth by 1 step.

Reply:
Makes sense!